### ISSUE TITLE
Add support for UILayoutGuide

### ISSUE THREAD

**Issue by Tricertops** [Tricertops/KeepLayout#72]
Now we have several classes that can participate in layout constraints:
- `UIView` – already supported, of course.
- `UILayoutAnchor` – makes little sense to support, since it’s an alternative mechanism.
- **`UILayoutGuide`** – the same layout behavior as `UIView`.

Those are two classes that share layout attributes and behavior. How to generalize this?

**Comment by iMartinKiss** [comment]
My first idea is to create a class, let’s say `KeepLayout`, where we move all the existing UIView methods. `UIView` and `UILayoutGuide` will then have a single property:

```objc
@property (readonly) KeepLayout *keep;
```

This wil allow mixed syntax with a single shared implementation:

```objc
view.keep.leftOffsetTo(view).equal = 10
guide.keep.leftOffsetTo(guide).equal = 10
view.keep.leftOffsetTo(guide).equal = 10
guide.keep.leftOffsetTo(view).equal = 10
```

**Comment by Tricertops** [comment]
This would nicely fit with the `NSArray` interface we currently have.

**Comment by Tricertops** [comment]
Partial support (hacked on existing API) was implemented in #87 that allows constructing constraints to Safe Area Layout Guides.

I have no further plans for API redesign. As time goes on, the built-in UIKit API is getting better.
